### PR TITLE
Updating rust version to 1.36.0

### PR DESCRIPTION
**Pull request by plauche**
- Updating our rust version to 1.36.0 in `rust_toolchain` and Docker. I've been running into compile issues with dependencies of `reqwest` with rustc 1.32.0.
- Cleans up new and lingering warnings from build/clippy/test

### CODE REVIEW

**Comment by plauche** [comment]
I'll need to regenerate Docker with this change before this PR will pass CI.

File: services/mai400-service/src/tests/mod.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use mai400_api::mock::*;
 use mai400_api::*;
 use std::sync::mpsc::channel;
 
+/// Generate new mock mai service
```
Comment (CatGarab) [inline services/mai400-service/src/tests/mod.rs]: nit: mai -> MAI

File: apis/clyde-3g-eps-api/src/commands/board_status.rs [hunk 2]
```diff
@@ -14,19 +14,18 @@
...
 /// To retrieve the data that represents the status, the command 0x01 should be
 /// sent followed by 0x00. The meaning of each bit of the returned status bytes
 /// is shown below. Please note that Data[3] is the first byte returned from the
 /// EPS and Data[0] is the last.
+use bitflags::bitflags;
+use eps_api::{EpsError, EpsResult};
+use rust_i2c::Command;
 
-/// EPS Board Status Codes
+// EPS Board Status Codes
```
Comment (CatGarab) [inline apis/clyde-3g-eps-api/src/commands/board_status.rs]: nit: This doc comment should be kept?

File: apis/clyde-3g-eps-api/src/telemetry/reset.rs [hunk 3]
```diff
@@ -73,10 +73,10 @@ pub struct Data {
...
 }
 
-/// Each of these reset telemetry commands may return two bytes or four bytes,
-/// depending on whether or not a daughterboard exists. The first two bytes
-/// will always represent the motherboard, the second two will represent
-/// the daughterboard. All counters roll over at 255 to 0.
+// Each of these reset telemetry commands may return two bytes or four bytes,
+// depending on whether or not a daughterboard exists. The first two bytes
+// will always represent the motherboard, the second two will represent
+// the daughterboard. All counters roll over at 255 to 0.
```
Comment (CatGarab) [inline apis/clyde-3g-eps-api/src/telemetry/reset.rs]: want to keep the doc comment

File: services/mai400-service/src/tests/mod.rs [hunk 4]
```diff
@@ -71,6 +72,7 @@ macro_rules! service_new {
     }};
 }
 
+/// Generate new mock mai service with read thread
```
Comment (CatGarab) [inline services/mai400-service/src/tests/mod.rs]: nit: MAI

**Comment by plauche** [comment]
[Upgrade rust version](https://trello.com/c/RyaT9lDc/1059-upgrade-rust-version)

File: apis/clyde-3g-eps-api/src/commands/board_status.rs (hunk 2 above)
Comment (plauche) [reply apis/clyde-3g-eps-api/src/commands/board_status.rs]: rustdoc-macro
It looks like `rustdoc` no longer (or never did?) uses that docstring.

File: apis/clyde-3g-eps-api/src/telemetry/reset.rs (hunk 3 above)
Comment (plauche) [reply apis/clyde-3g-eps-api/src/telemetry/reset.rs]: `rustdoc` also complained that this was an unused doc comment and that documentation would not be generated here.

Comment (CatGarab) [reply apis/clyde-3g-eps-api/src/telemetry/reset.rs]: Ah, ok. Can you either move or copy this comment section to be where `/// Reset Telemetry Variants` is? I think it's useful information and that's where the doc [section header](https://docs.kubos.com/1.16.0/rust-docs/clyde_3g_eps_api/ResetTelemetry/enum.Type.html) is getting generated

**Comment by plauche** [comment]
New docker has been pushed up.

**Pull request merged**